OID4VPVerifierTestBase failure in CI

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

testsuite

### Describe the bug

The test `OID4VPVerifierTestBase` is failing in main. It was triggered by commit 901b343b9c60d859b2b304652b4baaada51968dc for issue https://github.com/keycloak/keycloak/issues/51176.

Errors:

```
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   OID4VPDirectPostJwtTest.directPostRejectsContentEncryptionOutsideHaipSet:160->OID4VPVerifierTestBase.issueCredential:105 Credential request error: Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed. ==> expected: <null> but was: <Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed.>
[ERROR]   OID4VPDirectPostJwtTest.directPostRejectsPresentationWithWrongNonce:174->OID4VPVerifierTestBase.issueCredential:105 Credential request error: Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed. ==> expected: <null> but was: <Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed.>
[ERROR]   OID4VPDirectPostJwtTest.directPostRejectsResponseEncryptedToWrongKey:143->OID4VPVerifierTestBase.issueCredential:105 Credential request error: Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed. ==> expected: <null> but was: <Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed.>
[ERROR]   OID4VPDirectPostJwtTest.directPostRejectsUnencryptedPresentation:96->OID4VPVerifierTestBase.issueCredential:105 Credential request error: Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed. ==> expected: <null> but was: <Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed.>
[ERROR]   OID4VPDirectPostJwtTest.directPostRejectsUnknownEncryptionKey:186->OID4VPVerifierTestBase.issueCredential:105 Credential request error: Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed. ==> expected: <null> but was: <Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed.>
[ERROR]   OID4VPDirectPostJwtTest.happyFlowDecryptsPresentationAndContinuesLogin:84->OID4VPVerifierTestBase.issueCredential:105 Credential request error: Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed. ==> expected: <null> but was: <Signing of credential failed: HAIP-6.1.1 violation: signing certificate MUST NOT be self-signed.>
[INFO] 
[ERROR] Tests run: 11, Failures: 6, Errors: 0, Skipped: 0

```

### Version

999.0.0-SNAPSHOT

### Regression

- [ ] The issue is a regression

### Expected behavior

The test should work OK.

### Actual behavior

Test fails like in https://github.com/rmartinc/keycloak/actions/runs/30247319806/job/89935926840

### How to Reproduce?

```
cd keycloak
mvn clean install -DskipTests
cd tests/base
mvn test -Dtest=OID4VPDirectPostJwtTest
``` 

### Anything else?

_No response_

--- a/tests/base/src/test/java/org/keycloak/tests/oid4vc/presentation/OID4VPVerifierTestBase.java
+++ b/tests/base/src/test/java/org/keycloak/tests/oid4vc/presentation/OID4VPVerifierTestBase.java
@@ -20,7 +20,11 @@ package org.keycloak.tests.oid4vc.presentation;
 import java.io.IOException;
 import java.net.URLEncoder;
 import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
 import java.security.cert.X509Certificate;
+import java.security.spec.ECGenParameterSpec;
+import java.util.Base64;
 import java.util.List;
 import java.util.Map;
 import java.util.regex.Matcher;
@@ -30,7 +34,10 @@ import jakarta.ws.rs.core.Response;
 
 import org.keycloak.broker.oid4vp.OID4VPIdentityProviderConfig;
 import org.keycloak.broker.oid4vp.OID4VPIdentityProviderFactory;
+import org.keycloak.common.crypto.CryptoConstants;
+import org.keycloak.common.util.CertificateUtils;
 import org.keycloak.common.util.MultivaluedHashMap;
+import org.keycloak.common.util.PemUtils;
 import org.keycloak.crypto.Algorithm;
 import org.keycloak.crypto.KeyType;
 import org.keycloak.crypto.KeyUse;
@@ -86,17 +93,44 @@ public abstract class OID4VPVerifierTestBase extends OID4VCIssuerTestBase {
         keyProvider.setParentId(testRealm.getId());
         keyProvider.setProviderId(GeneratedEcdsaKeyProviderFactory.ID);
         keyProvider.setProviderType(KeyProvider.class.getName());
-        keyProvider.setConfig(new MultivaluedHashMap<>(Map.of(
+
+        try {
+            KeyPairGenerator kpg = KeyPairGenerator.getInstance(KeyType.EC);
+            ECGenParameterSpec ecSpec = new ECGenParameterSpec(CryptoConstants.EC_KEY_SECP256R1);
+            kpg.initialize(ecSpec);
+            KeyPair caKeyPair = kpg.generateKeyPair();
+            X509Certificate caCert = CertificateUtils.generateV1SelfSignedCertificate(caKeyPair, "Test CA");
+
+            KeyPair leafKeyPair = kpg.generateKeyPair();
+
+            X509Certificate leafCert = CertificateUtils.generateV3Certificate(
+                    leafKeyPair,
+                    caKeyPair.getPrivate(),
+                    caCert,
+                    "TestKey"
+            );
+
+            keyProvider.setConfig(new MultivaluedHashMap<>(Map.of(
                 Attributes.PRIORITY_KEY, List.of("100"),
                 Attributes.ENABLED_KEY, List.of("true"),
                 Attributes.ACTIVE_KEY, List.of("true"),
+                GeneratedEcdsaKeyProviderFactory.ECDSA_PRIVATE_KEY_KEY, List.of(
+                        Base64.getEncoder().encodeToString(leafKeyPair.getPrivate().getEncoded())),
+                GeneratedEcdsaKeyProviderFactory.ECDSA_PUBLIC_KEY_KEY, List.of(
+                        Base64.getEncoder().encodeToString(leafKeyPair.getPublic().getEncoded())),
                 GeneratedEcdsaKeyProviderFactory.ECDSA_ELLIPTIC_CURVE_KEY, List.of("P-256"),
-                Attributes.EC_GENERATE_CERTIFICATE_KEY, List.of("true"))));
+                Attributes.CERTIFICATE_KEY, List.of(PemUtils.encodeCertificate(leafCert)),
+                Attributes.ALGORITHM_KEY, List.of(Algorithm.ES256),
+                Attributes.KEY_USE, List.of(KeyUse.SIG.name()))
+            ));
 
-        try (Response response = testRealm.admin().components().add(keyProvider)) {
-            Assertions.assertEquals(201, response.getStatus(), "Failed to add the verifier signing key");
-            String location = response.getHeaderString("Location");
-            ecKeyComponentId = location.substring(location.lastIndexOf('/') + 1);
+            try (Response response = testRealm.admin().components().add(keyProvider)) {
+                Assertions.assertEquals(201, response.getStatus(), "Failed to add the verifier signing key");
+                String location = response.getHeaderString("Location");
+                ecKeyComponentId = location.substring(location.lastIndexOf('/') + 1);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to create HAIP-compliant SD-JWT signing key provider", e);
         }
     }
 

